crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -170,10 +170,6 @@ module sui::sui_system {
         commission_rate: u64,
         ctx: &mut TxContext,
     ) {
-        assert!(
-            validator_set::next_epoch_validator_count(&self.validators) < self.parameters.max_validator_candidate_count,
```

Comment:
Are you comfortable with us removing this check @lxfind? The upper bound is going to be auto bounded by the min stake requirement.

Reply:
I'm ok with it, but I would want to double check with the team that we are ok with the theoretical upper bound of 400 validators.

Reply:
We probably want to enforce the min stake requirement before we remove this right?

Reply:
Fair points. We should do both then:
- ask the team if we are okay with the theoretical upper bound of 400 validators.
- keep this line here and only remove it after we complete the join validator set flow where min stake is enforced. @longbowlu

Reply:
I added back the check